20231018-access-logs.md

```diff
@@ -0,0 +1,188 @@
+| Access Key ID | The S3 access key ID used to authenticate the request. | juudmr5vp28bjrqhny7luyibs9pq |  |
+| Macaroon Head | The hex-encoded macaroon head of the API key. | 70d745b9cb906bb75b4979a8160b8e8bef8e9d6b10d64d6ecd0f17efa02d649e |  |
+| Satellite Address | The Node URL of the satellite serving the request. | 12EayRS2V1kEsWESU9QMRseFhdxYxKicsiFmxrsLZHeLUtdps3S@us1.storj.io:7777 |  |
+| Request ID | The unique identifier of the request assigned by Storj | cgPYJruaG9W | Can be used to correlate with satellite access logs and Storj internal logs. |
+| Range Bytes | The value of the Range header if present. | bytes=0-8000000 |  |
+| Host Header | The endpoint used to connect to the Storj S3 compatible gateway. | https://gateway.storjshare.io |  |
+| Total Time | The number of milliseconds that the request was in flight from the server's perspective. This value is measured from the time that your request is received to the time that the last byte of the response is sent. Measurements made from the client's perspective might be longer because of network latency. | 70 |  |
+| Time to First Byte | The number of milliseconds that Storj spent processing your request. This value is measured from the time that the last byte of your request was received until the time that the first byte of the response was sent. | 10 |  |
+| Location | The datacenter code where the edge server is located. | TR2 | This is used for troubleshooting problems such as latency issues. |
+| Origin | The value of the HTTP Origin header, if present. | https://www.example.com |  |
+
+#### Storj bucket as the Loki Tenant ID
+
+Grafana Loki is [multi-tenant](https://grafana.com/docs/loki/latest/operations/multi-tenancy/). The logging agent can annotate the log entries with a Tenant ID while scraping them. The Loki server receives the annotated logs and aggregates them in different streams.
+
+As we want a separate log stream for each Storj bucket, it is natural to set the Loki Tenant ID to something uniquely identifying a Storj bucket.
+
+Storj bucket names are unique within a Storj project but not across different Storj projects. The Storj project public ID is a UUID - unique across all Storj projects, accounts, and satellites. Therefore, concatenating the bucket's project public ID and the bucket name is a unique identifier to use as the Loki Tenant ID.
+
+Examples:
+- `ef1b109249684b3e8466a865ce22a262_test-bucket`
+- `033f72b1cb4e437096fa238bf6b8502f_prod`
+
+#### Scraping and annotating access logs
+
+Three services generate access logs: Linksharing, Gateway-MT, and the Satellite API Pods. Each of these services has multiple instances. A logging agent will run next to each instance and scrape the generated access logs. The logging agent may filter the log entries or re-arrange their log format, but its main task is to annotate the log entries with the Tenant ID and push them to the Grafana Loki destination.
```

Comment:
Will we end up with "duplicate" logs for linksharing and gateway requests? In other words, wont there be a satellite log for every linksharing/gateway request -- or is it somehow combined based on request id?

Reply:
There will be a different stream (or "label" in Grafana terms) of logs for Linksharing, Gateway-MT, and Satellite API. There should be a correlating log entry in the satellite stream with a matching request ID for each request in the edge services. However, this may not be the case in the future if we introduce edge caching.